pylops/waveeqprocessing/kirchhoff.py

```diff
@@ -288,81 +274,93 @@ def __init__(
         self.dt = t[1] - t[0]
         self.nt = len(t)
 
-        # store ix-iy locations of sources and receivers
-        dx = x[1] - x[0]
-        if self.ndims == 2:
-            self.six = np.tile((srcs[0] - x[0]) // dx, (nr, 1)).T.astype(int).ravel()
-            self.rix = np.tile((recs[0] - x[0]) // dx, (ns, 1)).astype(int).ravel()
-        elif self.ndims == 3:
-            # TODO: 3D normalized distances
-            pass
-
-        # compute traveltime
+        # store ix-iy locations of sources and receivers for aperture filter
         self.dynamic = dynamic
+        if self.dynamic:
+            dx = x[1] - x[0]
+            if self.ndims == 2:
+                self.six = (
+                    np.tile((srcs[0] - x[0]) // dx, (nr, 1)).T.astype(int).ravel()
+                )
+                self.rix = np.tile((recs[0] - x[0]) // dx, (ns, 1)).astype(int).ravel()
+            elif self.ndims == 3:
+                # TODO: 3D normalized distances
+                raise NotImplementedError("dynamic=True currently not available in 3D")
+
+        # compute traveltime and distances
         self.travsrcrec = True  # use separate tables for src and rec traveltimes
         if mode in ["analytic", "eikonal", "byot"]:
             if mode in ["analytic", "eikonal"]:
                 # compute traveltime table
                 (
                     self.trav_srcs,
                     self.trav_recs,
-                    self.dist_srcs,
-                    self.dist_recs,
+                    dist_srcs,
+                    dist_recs,
                     trav_srcs_grad,
                     trav_recs_grad,
                 ) = Kirchhoff._traveltime_table(z, x, srcs, recs, vel, y=y, mode=mode)
                 if self.dynamic:
                     # need to add a scalar in the denominator of amplitude term to avoid
-                    # division by 0, currently set to 1/100 of max distance to avoid having
-                    # to large scaling around the source. This number may change in future
+                    # division by 0, currently set to 1e-4 of max distance to avoid having
+                    # too large scaling around the source. This number may change in future
                     # or left to the user to define
                     epsdist = 1e-2
```

Comment:
Should this be set to 1e-4 as per the comment?

Reply:
This is indeed inconsistent. I honestly do not know what is best to do to avoid blow up near source/rec... my experiments suggest 1e-2 is a reasonable choice, for now I fix the comment, but if you have an idea how to do this better I am happy to amend it